spring-cloud-vault-core/src/main/java/org/springframework/cloud/vault/VaultClient.java

```diff
@@ -102,6 +124,94 @@ public VaultClientResponse write(URI uri, Object entity, VaultToken vaultToken)
 				createHeaders(vaultToken)));
 	}
 
+	/**
+	 * Initialize the vault using with {@code createKeys} and {@code requiredKeys}
+	 *
+	 * @param createKeys The number of keys available to unseal
+	 * @param requiredKeys The number of keys required to unseal
+	 * @return A {@link VaultInitializedResponse} containing the initialized state of vault.
+	 */
+	public VaultInitializedResponse initialize(int createKeys, int requiredKeys) {
```

Comment:
These features rather belong into the `VaultTemplate`. Let's focus on #24 and keep the other changes out of the PR (see explanation in the summary).